subsys/pm/policy.c

```diff
@@ -136,6 +137,12 @@ const struct pm_state_info *pm_policy_next_state(uint8_t cpu, int32_t ticks)
 	uint8_t num_cpu_states;
 	const struct pm_state_info *cpu_states;
 
+#if defined(CONFIG_PM_DEVICE) && defined(CONFIG_PM_NEED_ALL_DEVICES_IDLE)
```

Comment:
This pollutes CONFIG_PM code with PM_DEVICE related stuff, so to me this is a -1. As explained, there are other more precise means to solve this issue, that is, with state locking (maybe refining its implementation to accommodate new requirements). There's also no need to enable CONFIG_PM_DEVICE to modify the behavior of system PM, this is why we have policy APIs.

Reply:
@gmarull I have no knowledge of other SoCs/platforms, but Intel ISH's SoC core low-power mode and device/peripheral power do have relationship. When CPU goes into deep sleep, devices' power will be gated.
So ISH does need this change, appreciate your understanding.

Reply:
@gmarull Hi Gerald, just get your point. Previously I thought CONFIG_PM_DEVICE is a sub-config of CONFIG_PM, but no from subsystem/pm/Kconfig.
but this seems weird to me. Shouldn't Device PM be part of PM (Power Management)? @ceolin @nashif

Reply:
@gmarull  I am not following what you are objecting to exactly, the code has changed since this:

> This pollutes CONFIG_PM code with PM_DEVICE related stuff, so to me this is a -1